### PR TITLE
Magento#25669: fixed issue "health_check.php fails if any database cache engine configured"

### PR DESCRIPTION
**Pull request by andrewbess**
### Description (*)

This PR fixes the issue magento/magento2/issues/25669: health_check.php fails if any database cache engine configured.

### Fixed Issues (if relevant)

1. magento/magento2/issues/25669: health_check.php fails if any database cache engine configured

### Manual testing scenarios (*)

1. Create custom cache engine like it described in [devdocs](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/cache/partial-caching/database-caching.html)
2. Configure any cache to database in env.php (for example look at [Configuration examples](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/cache/partial-caching/database-caching.html#mage-cache-db-config))
3. Run ./bin/magento setup:di:compile and wait finishing
4. Run health_check.php

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by andrewbess** [comment]
https://prnt.sc/q1rujl
It was removing wrong commit where I pulled changes from 2.3-develop into issue-25669.

File: pub/health_check.php [hunk 1]
```diff
@@ -63,8 +63,10 @@
         }
         $cacheBackendClass = $cacheConfig[ConfigOptionsListConstants::CONFIG_PATH_BACKEND];
         try {
+            /** @var \Magento\Framework\App\Cache\Frontend\Factory $cacheFrontendFactory */
+            $cacheFrontendFactory = $objectManager->get(Magento\Framework\App\Cache\Frontend\Factory::class);
```
Comment (rogyar) [inline pub/health_check.php:67]: Hi @andrewbess. 

I agree that the proposed change is "cleaner" and more maintainable. However, it violates backward compatibility. If an existing solution relies on `$cacheConfig[ConfigOptionsListConstants::CONFIG_PATH_BACKEND]` configuration, we will break the mentioned system by introducing this sort of change.

So, I would recommend do not refactor this part in the way as you did.

File: lib/internal/Magento/Framework/Lock/Backend/Cache.php [hunk 2]
```diff
@@ -45,7 +45,7 @@ public function lock(string $name, int $timeout = -1): bool
      */
     public function unlock(string $name): bool
     {
-        return $this->cache->remove($this->getIdentifier($name));
+        return (bool)$this->cache->remove($this->getIdentifier($name));
```
Comment (rogyar) [inline lib/internal/Magento/Framework/Lock/Backend/Cache.php:48]: Could I kindly ask you to cover this part using a simple unit test? Just mock a behavior that caused an error previously for `$this->cache->remove($this->getIdentifier($name))` and make sure that there's no error after the proposed change.

Thank you!

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by engcom-Echo** [comment]
I will take care of test coverage.

File: lib/internal/Magento/Framework/Lock/Backend/Cache.php (hunk 2 above)
Comment (andrewbess) [reply lib/internal/Magento/Framework/Lock/Backend/Cache.php:48]: Hello @rogyar 
Tests have already created.
Thank you @engcom-Echo

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-6834](https://jira.corp.magento.com/browse/ENGCOM-6834) has been created to process this Pull Request

**Comment by engcom-Echo** [comment]
QA passed

**Comment by engcom-Echo** [comment]
:heavy_check_mark: QA passed

**Pull request merged**